From 22a8f83270e2408df225e775977933d172b48b59 Mon Sep 17 00:00:00 2001
From: Darren Kenny <darren.kenny@oracle.com>
Date: Thu, 5 Nov 2020 10:15:25 +0000
Subject: kern/efi: Fix memory leak on failure

Free the memory allocated to name before returning on failure.

Fixes: CID 296222

Signed-off-by: Darren Kenny <darren.kenny@oracle.com>
Reviewed-by: Daniel Kiper <daniel.kiper@oracle.com>

Patch-Name: 2021-02-security/019-kern-efi-Fix-memory-leak-on-failure.patch
---
 grub-core/kern/efi/efi.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/grub-core/kern/efi/efi.c b/grub-core/kern/efi/efi.c
index 0de1f9fad..d6965a345 100644
--- a/grub-core/kern/efi/efi.c
+++ b/grub-core/kern/efi/efi.c
@@ -402,6 +402,7 @@ grub_efi_get_filename (grub_efi_device_path_t *dp0)
 	    {
 	      grub_error (GRUB_ERR_OUT_OF_RANGE,
 			  "malformed EFI Device Path node has length=%d", len);
+	      grub_free (name);
 	      return NULL;
 	    }
 
